### PR TITLE
Added ability to set nodeSelector and serviceAccountName on the pod

### PR DESCRIPTION
**Pull request by billmcmath**
- Added the ability to set a service account name using the `--service-account-name` flag.
- Added the ability to set node selectors using the `--node-selector` flag.

### CODE REVIEW

File: cmd/root.go [hunk 1]
```diff
@@ -101,8 +104,19 @@ along with the synchronized files.`,
 
 		c := kubectl.NewClient(config)
 
+		nodeSelectors := make(map[string]string)
+		if len(opt.NodeSelectorKeys) > 0 {
+			if len(opt.NodeSelectorKeys) != len(opt.NodeSelectorValues) {
+				fmt.Fprintln(stderr, "Node selector keys and values don't match, dropping node selector. %d != %d", len(opt.NodeSelectorKeys), len(opt.NodeSelectorValues))
```
Comment (ernoaapa) [inline cmd/root.go]: I believe that in case of invalid parameters, the whole command should fail.
So return error here.

File: cmd/root.go [hunk 2]
```diff
@@ -188,6 +202,9 @@ func init() {
 	rootCmd.Flags().BoolVarP(&opt.TTY, "tty", "t", opt.TTY, "Stdin is a TTY")
 	rootCmd.Flags().StringSliceVar(&opt.Includes, "include", []string{}, "Include only specific paths from current directory for syncing")
 	rootCmd.Flags().StringSliceVar(&opt.Excludes, "exclude", []string{}, "Exclude only specific paths from current directory for syncing")
+	rootCmd.Flags().StringVar(&opt.ServiceAccountName, "service-account-name", opt.ServiceAccountName, "The service account name that you want the pod to use")
+	rootCmd.Flags().StringSliceVar(&opt.NodeSelectorKeys, "node-selector-keys", []string{}, "The node selector keys that you want applied")
+	rootCmd.Flags().StringSliceVar(&opt.NodeSelectorValues, "node-selector-values", []string{}, "The node selector values that you want applied")
```
Comment (ernoaapa) [inline cmd/root.go]: Can you use `StringToString` flag? https://godoc.org/github.com/spf13/pflag#StringToString

Comment (billmcmath) [reply cmd/root.go]: Thanks, I was looking for a function like this, I don't know how I missed it.

**Comment by billmcmath** [comment]
I've tested this with no service account name and with a service account name ` --service-account-name "dev-sa"` with the serviceAccountName, it uses the value provided, with no service account name it falls back to default.  I've also tested with 0, 1 and 2 uses of `--node-selector="nodeEnvironment=dev" --node-selector="nodeRegion=west"`, with none the option is skipped, with one value that value is listed and with multiple values all values are listed.

File: cmd/root.go (hunk 1 above)
Comment (billmcmath) [reply cmd/root.go]: With the change you suggested below, this block of code has been removed.

**Comment by ernoaapa** [comment]
Thank you @billmcmath! 👍🏻

**Pull request merged**